### ISSUE TITLE
Support "false" value for Region#show() options triggerBeforeAttach/triggerAttach

### ISSUE THREAD

**Issue by ianmstew** [marionettejs/backbone.marionette#2484]
The Region gives the developer power to use `show()` option `{triggerBeforeAttach: true}` but not `{triggerBeforeAttach: false}`, and similarly with `triggerAttach`.

While the ability to use `show()` options set to `false` is not documented one way or another, Backbone `Collection#set()` methods establish a precedent of [disabling a default behavior with `false` method options](http://backbonejs.org/#Collection-set).

In this example, onBeforeAttach will fire, [as documented](http://marionettejs.com/docs/marionette.region.html#onbeforeattach-onattach):

```js
var FooRegion = Marionette.Region.extend({
  triggerBeforeAttach: false
});
region.show(view, {triggerBeforeAttach: true});
```

In this example, onBeforeAttach will _also_ fire (no documentation for this case):

```js
var FooRegion = Marionette.Region.extend({
  triggerBeforeAttach: true  // The default value, but presented here for clarity
});
region.show(view, {triggerBeforeAttach: false});
```

The second example does not match my expectations, partly set up by Backbone, because the explicit `false` option is ignored.

The fix for this is small (two lines).  The question is whether we want to support explicit false options, particularly in this case.

**Comment by jasonLaster** [comment]
@ianmstew nice find. I think we should support this, I'd definitely expect that too work

**Comment by ahumphreys87** [comment]
:+1: tagging for 2.4.2

**Comment by ianmstew** [comment]
Nice.  On it!

**Comment by ahumphreys87** [comment]
Fixed with #2529
